src/main/java/org/springframework/cloud/scheduler/spi/kubernetes/KubernetesSchedulerPropertyResolver.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.scheduler.spi.kubernetes;
+
+import org.springframework.cloud.scheduler.spi.core.ScheduleRequest;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Provides centralized lookup methods for properties that can resolved from multiple
+ * sources along with any custom logic to be applied.
+ *
+ * @author Chris Schaefer
+ */
+class KubernetesSchedulerPropertyResolver {
+	private static final NestedCommaDelimitedVariableParser nestedCommaDelimitedVariableParser
+			= new NestedCommaDelimitedVariableParser();
+
+	public static String getImagePullPolicy(ScheduleRequest scheduleRequest,
+			KubernetesSchedulerProperties kubernetesSchedulerProperties) {
+		String imagePullPolicy = scheduleRequest.getSchedulerProperties()
+				.get(KubernetesSchedulerProperties.KUBERNETES_SCHEDULER_PROPERTIES + ".imagePullPolicy");
+
+		if (StringUtils.hasText(imagePullPolicy)) {
+			return ImagePullPolicy.relaxedValueOf(imagePullPolicy).name();
+		}
+
+		return kubernetesSchedulerProperties.getImagePullPolicy().name();
+	}
+
+	public static EntryPointStyle getEntryPointStyle(ScheduleRequest scheduleRequest,
+			KubernetesSchedulerProperties kubernetesSchedulerProperties) {
+		String entryPointStyle = scheduleRequest.getSchedulerProperties()
+				.get(KubernetesSchedulerProperties.KUBERNETES_SCHEDULER_PROPERTIES + ".entryPointStyle");
+
+		if (StringUtils.hasText(entryPointStyle)) {
+			return EntryPointStyle.relaxedValueOf(entryPointStyle);
+		}
+
+		return kubernetesSchedulerProperties.getEntryPointStyle();
+	}
+
+	public static Map<String, String> getTaskEnvironmentVariables(ScheduleRequest request,
+			KubernetesSchedulerProperties kubernetesSchedulerProperties) {
+		Map<String, String> environmentVariableMap = new HashMap<>();
+
+		String taskEnvironmentVariables = request.getSchedulerProperties()
+				.get(KubernetesSchedulerProperties.KUBERNETES_SCHEDULER_PROPERTIES + ".environmentVariables");
+
+		environmentVariableMap.putAll(parseEnvironmentVariables(taskEnvironmentVariables));
```

Comment:
Wouldn't you want the taskEnvironment variables to be put after the KubernetesSchedulerProperties in the case their maybe an override from the user creating the schedule?

Reply:
yup makes sense